ts/parts/Tick.ts

```diff
@@ -360,6 +357,159 @@ H.Tick.prototype = {
         }
     },
 
+    /**
+     * Try to replace the label if the same one already exists.
+     *
+     * @private
+     * @function Highcharts.Tick#moveLabel
+     * @param {string} str
+     * @param {Highcharts.XAxisLabelsOptions} labelOptions
+     *
+     * @return {void}
+     */
+    moveLabel: function (
```

Comment:
Could you explain what is wrong in your test? Categories need to be set by `point.name` because otherwise they are not related to points and their order does not change: http://jsfiddle.net/BlackLabel/5dp1swrf/

Reply:
Ok, it seems I pasted the wrong URL. This is what I meant: http://jsfiddle.net/highcharts/d3q8g1pu/7/

Reply:
Ok, now I see your idea. The problem is that `setCategories` fires `axis.update`, which destroys an axis and initiates it again.

https://github.com/highcharts/highcharts/blob/4605ae2c31481deb6dda80de461572ca9f13ebbd/ts/parts/Dynamics.ts#L1815-L1821

We can store old labels and try to use them again, but I think it may require a lot of new code and changes. What do you think @TorsteinHonsi ?

Reply:
Yes I agree.